src/middlewares/checkToken.ts

```diff
@@ -0,0 +1,25 @@
+import { Response, NextFunction } from 'express';
+import { templateErrors } from '../helpers';
+import { verifyToken } from '../helpers/jwt';
+import { Decode, RequestWithUserRole } from '../types';
+
+const checkTokenForLogin = () => async (
```

Comment:
```suggestion
const checkTokenForLogin = async (
```
No need to declare a function that return express middleware, we did that in checkAuth cause we need to call it with different roles
 ```js
 checkAuth('therapist');
checkAuth('user');
```
and checkAuth was a function that return a middleware, But here we can do the follow
```js
const checkToken = (req, res, next) => {}
// and call it 
route(checkToken, ourController)
```